### DISCUSSION TITLE
How do I match a text node with a colon in it

### DISCUSSION THREAD

**Discussion by Hubro** [tree-sitter/tree-sitter#1443] [Q&A] [answered]
I'm trying to write some Tree-sitter queries for [nvim-gps](https://github.com/SmiteshP/nvim-gps).

Most of my queries look like this, and they work fine:

```scm
...

((statement
	(statement_keyword "list")
	(argument) @list-name) @scope-root)

((statement
	(statement_keyword "leaf-list")
	(argument) @leaf-list-name) @scope-root)

((statement
	(statement_keyword "leaf")
	(argument) @leaf-name) @scope-root)

...
```

However when I try to match a text node with a colon symbol in it:

```scm
((statement
	(statement_keyword "tailf:action")
	(argument) @action-name) @scope-root)
```

I get this error:

> query: invalid node type at position 770

Why doesn't this work? Is there some escape syntax I can use to match this text node?

**Comment by ahelwer** [comment]
Is `"tailf:action"` an explicitly enumerated choice of terminal in your grammar, or is it matched with a regex or some combination of nodes?

**Reply by Hubro** [reply]
A regex, I believe.

**Reply by ahelwer** [reply, marked as answer]
You can only use the `"text"` thing to match invisible nodes that are created from explicit terminals, like keywords and such. If the terminal is matched with a regex you have to use a predicate. This is because tree-sitter doesn't store the actual text being parsed in its tree, only byte offsets. With explicit terminals the node name is (usually, subject to aliasing) the actual text being matched. But with a regex it will just be a node containing byte offsets for the thing it matched; you'll need to pass this to a predicate to match the actual text.
